### DISCUSSION TITLE
/sitemap/[slug] and sitemap.ts conflicts in Nextjs15 (app router)

### DISCUSSION THREAD

**Discussion by timkinsman** [vercel/next.js#74055] [Help]
### Summary

Hey, when migrating from next14 to next15 (both app router). It seems that having a page for `app/sitemap/[slug]/page.tsx` (a HTML sitemap) and `app/sitemap.ts` (a XML sitemap) causes errors. Is this intentional? I'm aware that a related change was posted in the blog.

- [Breaking] Remove .xml extension for dynamic sitemap routes and align sitemap URLs between development and production ([PR](https://github.com/vercel/next.js/pull/65507))

However, I can't seem to wrap my head around it, is the above scenario just not achievable now?

### Additional information

```js
Error: You cannot use different slug names for the same dynamic path ('slug' !== '__metadata_id__').
    at Array.forEach (<anonymous>)
 ⨯ unhandledRejection: Error: You cannot use different slug names for the same dynamic path ('slug' !== '__metadata_id__').
    at Array.forEach (<anonymous>)
 ⨯ unhandledRejection:  Error: You cannot use different slug names for the same dynamic path ('slug' !== '__metadata_id__').
    at Array.forEach (<anonymous>)
```

### Example

https://github.com/timkinsman/nextjs15-bugrepo-sitemap-conflict

**Comment by smoglica** [comment]
I'm experiencing the same issue. I'm using Next.js `15.1.6`. I had no problems when using `next dev`, if I use turbopack with `next dev --turbo` I get:

![image](https://github.com/user-attachments/assets/103171bc-62e1-454f-848d-2cc037fceec9)

**Reply by akkis** [reply]
Same issue with 15.2.2. --turbopack parameter needs to be removed from dev script on package.json file.
